docs/WASI-api.md

```diff
@@ -406,16 +407,16 @@ Inputs:
 
     The file descriptor from which to read data.
 
-- <a href="#fd_pread.iovs" name="fd_pread.iovs"></a><code>const [\_\_wasi\_iovec\_t](#iovec) \*<strong>iovs</strong></code> and <a href="#fd_pread.iovs_len" name="fd_pread.iovs_len"></a><code>size\_t <strong>iovs\_len</strong></code>
-
-    List of scatter/gather vectors in which to store data.
-
 - <a href="#fd_pread.offset" name="fd_pread.offset"></a><code>[\_\_wasi\_filesize\_t](#filesize) <strong>offset</strong></code>
 
     The offset within the file at which to read.
 
 Outputs:
 
+- <a href="#fd_pread.iovs" name="fd_pread.iovs"></a><code>const [\_\_wasi\_iovec\_t](#iovec) <strong>iovs</strong></code> and <a href="#fd_pread.iovs_len" name="fd_pread.iovs_len"></a><code>size\_t <strong>iovs\_len</strong></code>
+
+    List of scatter/gather vectors in which to store data.
```

Comment:
This change is not correct. We don't have a powerful vocabulary for describing this yet, so right now, saying that the function takes in a pointer to an array of records that point to buffers that it modifies as a side effect at least allows us to correctly describe its signature.